neoload/commands/report.py

```diff
@@ -0,0 +1,1057 @@
+#     "Zone": "Default zone"
+#   }
+    perc_fields = list(filter(lambda x: x.startswith('percentile'), json_values.keys()))
+    convert_to_seconds = ['minDuration','maxDuration','sumDuration','avgDuration'] \
+                        + perc_fields
+
+    convert_element_fields_to_seconds(convert_to_seconds,json_values)
+
+    round_fields = convert_to_seconds \
+                    + ['successRate','failureRate']
+
+    round_element_fields(round_fields,json_values,3)
+
+    el["display_name"] = full_name
+    el["parent"] = parent
+    el["user_path"] = user_path
+    el["aggregate"] = json_values
+    el["points"] = json_points
+    el["raw"] = json_raws
+    el["totalCount"] = el["aggregate"]["successCount"] + el["aggregate"]["failureCount"]
+    if el["totalCount"] == 0:
+        el["successRate"] = 0
+        el["failureRate"] = 0
+    else:
+        el["successRate"] = el["aggregate"]["successCount"] / el["totalCount"]
+        el["failureRate"] = el["aggregate"]["failureCount"] / el["totalCount"]
+
+    return el
+
+def get_element_full_name(el):
+    return el['name'] if not 'path' in el else " \\ ".join(el['path'])
+
+def get_element_data_from_sources(include_points, time_binding, use_txn_raw, result_id, el, statistics_list):
+    viable_raw_element = (use_txn_raw and el['type'] in ["TRANSACTION"])
+    json_raws = []
+    json_points = []
+    if (include_points or time_binding is not None):
+        if viable_raw_element:
+            logging.getLogger().debug("Starting get_element_data_from_sources for element ({})".format(el['id']))
+            json_raws = rest_crud.get(get_end_point(result_id, __operation_elements) + "/" + el['id'] + "/raw?format=JSON")
+        else:
+            json_points = rest_crud.get(get_end_point(result_id, __operation_elements) + "/" + el['id'] + "/points?statistics=" + ",".join(statistics_list))
+
+    if not time_binding is None:
+        json_points = filter_by_time(json_points, time_binding, lambda p: int(p['from'])/1000, lambda p: int(p['to'])/1000)
+        json_raws = filter_by_time(json_raws, time_binding, lambda p: p['Elapsed']/1000, lambda p: p['Elapsed']/1000)
+
+    return (json_raws,json_points)
+
+def get_element_data_by_raws(json_raws,json_values):
+    field_name = 'Response time'
+    perc_points = list(sorted(map(lambda x: x[field_name], json_raws)))
+    sum_of_count = len(json_raws)
+    sum_of_errors = 0 if len(json_raws) < 1 else len(list(filter(lambda x: x['Success'] not in ["yes"],json_raws)))
+    json_values['minDuration'] = 0 if len(json_raws) < 1 else min(list(map(lambda x: x[field_name], json_raws)))
+    json_values['maxDuration'] = 0 if len(json_raws) < 1 else max(list(map(lambda x: x[field_name], json_raws)))
+    json_values['avgDuration'] = 0 if len(json_raws) < 1 else statistics.mean(list(map(lambda x: x[field_name], json_raws)))
+    return (perc_points,sum_of_count,sum_of_errors)
+
+def get_element_data_by_points(json_points,json_values):
```

Comment:
Percentiles for tests without raw data are computed based on points get from the API. But for this long test (it lasts 7 days), computed percentiles are far from NLW values! See this comparison :
![image](https://user-images.githubusercontent.com/57721694/107934748-bbab6100-6f80-11eb-8f9c-2bfc12421ca4.png)

Reply:
This is a known limitation on the aggregation when the report is filtered by time AND raw datas are not available. This is specified by a warning in transactions table in the html report.